src/my-server.js

```diff
@@ -9,18 +14,26 @@ class Server {
 
     start () {
         return new Promise((resolve, reject) => {
-            this.server = this.app.listen(this.port, resolve).on('error', reject);
+            this.server = this.app.listen(this.port, resolve)
+                .once('error', err => {
+                    error(err);
+                    reject();
+                });
         })
-            .then(() => console.log(`Server started on ${this.host}:${this.port}`))
-            .catch(() => console.log(`Server failed to start`));
+            .then(() => log('Server started on %s:%d', this.host, this.port))
+            .catch(() => error('Server failed to start'));
```

Comment:
The problem here is that a code that calls this function won't receive the exception and won't be aware if something goes wrong. We should output the message and re-reject the exception to pass it to the caller.